server/src/main/java/org/elasticsearch/index/mapper/DocumentParser.java

```diff
@@ -451,28 +453,7 @@ private static void parseObjectDynamic(DocumentParserContext context, String cur
                 dynamicObjectMapper = DynamicFieldsBuilder.createDynamicObjectMapper(context, currentFieldName);
                 context.addDynamicMapper(dynamicObjectMapper);
             }
-            if (context.parent().subobjects() == false) {
-                if (dynamicObjectMapper instanceof NestedObjectMapper) {
-                    throw new DocumentParsingException(
-                        context.parser().getTokenLocation(),
-                        "Tried to add nested object ["
-                            + dynamicObjectMapper.simpleName()
-                            + "] to object ["
-                            + context.parent().name()
-                            + "] which does not support subobjects"
-                    );
-                }
-                if (dynamicObjectMapper instanceof ObjectMapper) {
-                    throw new DocumentParsingException(
-                        context.parser().getTokenLocation(),
-                        "Tried to add subobject ["
-                            + dynamicObjectMapper.simpleName()
-                            + "] to object ["
-                            + context.parent().name()
-                            + "] which does not support subobjects"
-                    );
-                }
-            }
```

Comment:
can you expand on removing these?

Reply:
Actually wasn't necessary, I added it back! Was needed in a previous implementation I tired and newbiely forgot to add it back... I'll be more careful next times